ffi/derive/src/impl_visitor.rs

```diff
@@ -287,8 +295,12 @@ impl<'ast> Visit<'ast> for FnVisitor<'ast> {
     fn visit_impl_item_method(&mut self, node: &'ast syn::ImplItemMethod) {
         self.doc = find_doc_attr(&node.attrs).cloned();
 
-        if !matches!(node.vis, syn::Visibility::Public(_)) {
-            abort!(node.vis, "Methods defined in the impl block must be public");
+        // NOTE: only public method in inherit impls are allowed
+        if self.trait_name.is_none() && !matches!(node.vis, syn::Visibility::Public(_)) {
+            abort!(
+                node.vis,
+                "Methods defined in the inherit impl block must be public"
```

Comment:
```suggestion
                "Methods defined in the inherent `impl` block must be public"
```

On another note, I'd actually prefer if you just skipped them, instead of failing to compile. Notify via https://docs.rs/proc-macro-error/latest/proc_macro_error/macro.emit_warning.html

Reply:
We are `abort`ing execution all over the place (i mean `ffi_export/import` macros) where we don't support such item. 
What benefits will we get if we skip private functions?

Reply:
Basically, you wouldn't force the user to add the function elsewhere. A private function is perfectly valid code, while it shouldn't be reflected in the ffi export, forcing people to have multiple inherent impls is not good design.

Reply:
Ok, in this case we should make `impl` blocks more permissive in general, skip consts, types, ... instead of aborting execution.

Reply:
exactly.

Reply:
Now private functions are skipped, however i left this abortion, to indicate that something going wrong if we try to export private function.